File: hw3/hw3 - vectors/driver2.cpp
#include "tetris.h"
#include "tetromino.h"

int main(){

	int height, width;
	char shape;
	bool gameOver = false;
	vector<Tetromino> tetrominos;

	cout << "--------- Welcome to TETRIS!! ---------" << endl << endl;
	cout << "Enter the height of the board: ";
	cin >> height;
	cout << "Enter the width of the board: ";
	cin >> width;

	Tetris tetris(height, width);

	do{
		cout << "\033[s" << "\033[K";
		cout << "Enter the tetromino shape(I, O, T, J, L, S, Z  &  R for random, Q for quit): ";
		cin >> shape;
		shape = tetris.upperCase(static_cast<int>(shape));
		while(shape != 'I' && shape != 'O' && shape != 'T' && shape != 'J' && shape != 'L' && shape != 'S' && shape != 'Z' && shape != 'R' && shape != 'Q'){
			cout << "\033[u" << "\033[K";
			cout << "Enter the tetromino shape(I, O, T, J, L, S, Z  &  R for random, Q for quit): ";
			cin >> shape;
			shape = tetris.upperCase(static_cast<int>(shape));
		}
		if(shape == 'Q') break;
		if(shape == 'R') shape = tetris.randShape();
		tetrominos.push_back(Tetromino::defShape(shape));
		if(tetrominos.size() != 0) tetris += tetrominos[tetrominos.size() - 1];
		tetris.animate(tetrominos[tetrominos.size() - 1]);
		gameOver = tetris.isGameOver();
	}while(!gameOver);

	return 0;
}

File: hw7/Driver2.java
package tetris;

import java.util.ArrayList;
import java.util.Scanner;

public class Driver2 {

	public static void main(String[] args) throws InterruptedException {

		Scanner scanner = new Scanner(System.in);
		ArrayList<Tetromino> tetrominos = new ArrayList<>();
		
		int height, width;
		char shape;
		
		System.out.println("--------- Welcome to TETRIS!! ---------");
		System.out.println("Enter the height of the board: ");
		height = scanner.nextInt();
		System.out.println("Enter the width of the board: ");
		width = scanner.nextInt();

		Tetris tetris = new Tetris(height, width);
		boolean gameOver = false;

		do {
			
			System.out.println("Enter the tetromino shape(I, O, T, J, L, S, Z  &  R for random, Q for quit): ");
			shape = scanner.next().charAt(0);
			shape = Character.toUpperCase(shape);
			while (shape != 'I' && shape != 'O' && shape != 'T' && shape != 'J' && shape != 'L' && shape != 'S' && shape != 'Z' && shape != 'R' && shape != 'Q') {
				System.out.println("Enter the tetromino shape(I, O, T, J, L, S, Z  &  R for random, Q for quit): ");
				shape = scanner.next().charAt(0);
				shape = Character.toUpperCase(shape);
			}
			if (shape == 'Q') {
				break;
			}
			if (shape == 'R') {
				shape = Tetromino.randShape();
			}

			tetrominos.add(new Tetromino(shape));
			tetris.addTetromino(tetrominos.get(tetrominos.size() - 1));
			tetris.animate(tetrominos.get(tetrominos.size() - 1));

		} while (!gameOver);
		
		tetris.addTetromino(tetrominos.get(tetrominos.size() - 1));
		tetris.animate(tetrominos.get(tetrominos.size() - 1));

	}

}
